### PR TITLE
NAS-133125 / 25.04 / remove psutil from webshell_app.py

### PR DESCRIPTION
**Pull request by yocalebo**
This removes the final user of `psutil`. After investigation (and testing), I realized that someone had already added an `abort()` method to the shell worker thread. This is all that's needed to be called to ensure the `login` parent process AND the immediate child process (the shell) is terminated including all children below it.

I changed to using kill process group id so that it handles all dependents for us instead of generating a list of child processes unnecessarily.

### CODE REVIEW

File: src/middlewared/middlewared/apps/webshell_app.py [hunk 1]
```diff
@@ -200,7 +197,7 @@ def abort(self):
         asyncio.run_coroutine_threadsafe(self.ws.close(), self.loop)
 
         with contextlib.suppress(ProcessLookupError):
-            os.kill(self.shell_pid, signal.SIGTERM)
+            os.killpg(os.getpgid(self.shell_pid), signal.SIGTERM)
```
Comment (themylogin) [inline src/middlewared/middlewared/apps/webshell_app.py:200]: And if they don't respect `SIGTERM`?

Comment (yocalebo) [reply src/middlewared/middlewared/apps/webshell_app.py]: Then someone is doing something via the webshell we never designed it for and we've probably got other issues. However, it's a valid point so I extended `terminate_pid` and call that instead. Thanks!

File: src/middlewared/middlewared/utils/os.py [hunk 2]
```diff
@@ -42,12 +42,17 @@ def close_fds(low_fd, max_fd=None):
     closerange(low_fd, max_fd)
 
 
-def terminate_pid(pid: int, timeout: int = 10) -> bool:
-    # Send SIGTERM to request the process to terminate
-    kill(pid, SIGTERM)
+def terminate_pid(pid: int, timeout: int = 10, get_pgid: bool = False) -> bool:
```
Comment (themylogin) [inline src/middlewared/middlewared/utils/os.py:45]: Should it be named `use_pgid`? `get` sounds like we’ll return something. Also, this method now deserves a params doc.

Comment (yocalebo) [reply src/middlewared/middlewared/utils/os.py]: Sure, fixed it and added docstring.

**Pull request merged**